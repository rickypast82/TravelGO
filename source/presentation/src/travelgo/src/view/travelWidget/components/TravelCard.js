//@flow
import React from 'react'
import styled from 'styled-components'
import TravelImage from './TravelImage'
import { getResource } from '../../../lib/resource-utility/utils'
import { SMALL_SIZE, type SizeTravelWidget } from '../constants'

type Props = {|
    id:number,
    destination:string,
    departureDate?:number,
    returnDate?:number,
    note:string,
    cardSize?:SizeTravelWidget,
    imageSrc:string
|};

const SMALL_WIDTH = 25;
const MEDIUM_WIDTH = 35;
const SMALL_BODY_WIDTH = 20;
const MEDIUM_BODY_WIDTH = 29;

export const getWidth = (size?:SizeTravelWidget) =>{
    if(size)
        return size === SMALL_SIZE ? SMALL_WIDTH : MEDIUM_WIDTH;
    else
        return SMALL_WIDTH;
}

export const getWidthBody = (size?:SizeTravelWidget) =>{
    if(size)
        return size === SMALL_SIZE ? SMALL_BODY_WIDTH : MEDIUM_BODY_WIDTH;
    else
        return SMALL_BODY_WIDTH;
}

const Container = styled.div`
    width: ${(props:Props) => getWidth(props.cardSize) + 'em'};
`
const Header = styled.div`
    display:block;
`
const Body = styled.div`
    display:block;
    border-left: 1px solid rgba(0,0,0,.125);
    border-bottom: 1px solid rgba(0,0,0,.125);
    border-right: 1px solid rgba(0,0,0,.125);
    border-radius: 0px 0px 3px 3px;
    width: ${(props:Props) => getWidthBody(props.cardSize) + 'em'};
`
const Title = styled.div``
const Text = styled.div``

const Footer = styled.div``

const TravelCard = ({cardSize,destination,departureDate,imageSrc,note,returnDate}:Props) => {

    return( 
        <Container className='travelCardDivContainer' cardSize={cardSize}>
            <Header className='travelCardHeader'>
               <TravelImage imageSize={cardSize} src={getResource(imageSrc)} />
            </Header>
            <Body cardSize={cardSize} className='travelCardBody'>
                <Title>{destination} </Title>
                <Text> {note}</Text>
                <Text> {departureDate}</Text>
                <Text> {returnDate}</Text>
            </Body>
            <Footer className='travelCardFooter'/>
        </Container>
    )
};

export default TravelCard;