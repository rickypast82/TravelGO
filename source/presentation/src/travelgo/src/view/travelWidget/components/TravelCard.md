### TravelCard2 use gui components  ###

```js
<TravelCard2 id={1} 
    destination='Roma' 
    note='note roma' 
    departureDate={200} 
    returnDate={300}
    imageSrc='Roma.jpg'
    />
```
### TravelCard2 no departureDate, no returnDate ###

```js
<TravelCard2 id={1} 
    destination='Roma' 
    note='note roma' 
    imageSrc='Roma.jpg'
    />