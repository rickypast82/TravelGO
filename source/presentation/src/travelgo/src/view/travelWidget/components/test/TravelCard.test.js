//@flow
/*eslint no-magic-numbers: "off"*/
import React from 'react';
import { shallow } from 'enzyme';
import TravelCard from '../TravelCard';

describe('TravelCard components test', () => {
 
    it('renders without crashing with all props', () => {
        const wrapper = shallow( 
            <TravelCard 
                id={1}
                destination='Roma' 
                note='note roma' 
                departureDate={200} 
                returnDate={300}
                imageSrc='Roma.jpg'
            />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders without crashing with required props', () => {
        const wrapper = shallow( 
            <TravelCard 
                id={1}
                destination='Roma' 
                note='note roma' 
                imageSrc='Roma.jpg'
            />);
        expect(wrapper).toMatchSnapshot();
    });

});
