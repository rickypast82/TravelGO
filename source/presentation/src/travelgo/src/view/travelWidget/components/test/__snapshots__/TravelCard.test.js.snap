// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelCard components test renders without crashing with all props 1`] = `
<styled.div
  cardSize="MEDIUM@travelWidget"
  className="travelCardDivContainer"
>
  <styled.div
    className="travelCardHeader"
  >
    <styled.img
      imageSize="MEDIUM@travelWidget"
      src="Roma.jpg"
    />
  </styled.div>
  <styled.div
    cardSize="MEDIUM@travelWidget"
    className="travelCardBody"
  >
    <styled.div>
      Roma
       
    </styled.div>
    <styled.div>
       
      note roma
    </styled.div>
    <styled.div>
       
      200
    </styled.div>
    <styled.div>
       
      300
    </styled.div>
  </styled.div>
  <styled.div
    className="travelCardFooter"
  />
</styled.div>
`;

exports[`TravelCard components test renders without crashing with required props 1`] = `
<styled.div
  className="travelCardDivContainer"
>
  <styled.div
    className="travelCardHeader"
  >
    <styled.img
      src="Roma.jpg"
    />
  </styled.div>
  <styled.div
    className="travelCardBody"
  >
    <styled.div>
      Roma
       
    </styled.div>
    <styled.div>
       
      note roma
    </styled.div>
    <styled.div>
       
    </styled.div>
    <styled.div>
       
    </styled.div>
  </styled.div>
  <styled.div
    className="travelCardFooter"
  />
</styled.div>
`;
